game/src/main/java/io/castles/game/GameLobby.java

```diff
@@ -49,21 +46,17 @@ public void removePlayer(Player player) {
 
     public void removePlayer(UUID playerId) {
         Player playerToRemove = getPlayerById(playerId);
-        boolean removed = this.players.remove(playerToRemove);
-        if (!removed) {
-            throw new IllegalArgumentException(String.format("Player %s was not found in the list of players %s", playerToRemove, players));
-        }
+        removePlayer(playerToRemove);
     }
 
     private Player getPlayerById(UUID playerId) {
-         List<Player> matchingPlayers = this.players.stream()
-             .filter(player -> player.getId().equals(playerId))
-             .collect(Collectors.toList());
-
-         if (matchingPlayers.isEmpty())
-             throw new IllegalArgumentException(String.format("Player with if %s was not found in the list of players %s", playerId, players));
-
-         return matchingPlayers.get(0);
+        try {
+            return this.players.stream()
+                    .filter(player -> player.getId().equals(playerId))
+                    .findFirst().get();
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(String.format("Player with if %s was not found in the list of players %s", playerId, players));
+        }
```

Comment:
Is this usage of `.findFirst()` correct?

Reply:
Never saw it that way :) usually you work with the optional value by either calling `isPresent()` or chain an `orElseThrow` like I did in the following commit